database/rrdhost.c

```diff
@@ -707,14 +707,31 @@ void rrdhost_save_charts(RRDHOST *host) {
     rrdhost_unlock(host);
 }
 
+static int is_valid_label_key(char *key) {
+    while(key) {
+        if(!isdigit(*key) || !isalpha(*key) || *key != '.' || *key != '_' || *key != '-')
```

Comment:
the condition is wrong, have you tested it?

Reply:
> while(key) = > while (*key != '\0')

Reply:
And check for `NULL` is missing

Reply:
It should be:
```
if(!(*key == '.' || *key == '_' || *key == '-' || isdigit(*key) || isalpha(*key) ))
```
Please test it.

Reply:
I changed for your version, it is more readable, and I also adjust the loop end.